### DISCUSSION TITLE
Saving an array in json field

### DISCUSSION THREAD

**Discussion by ghost** [pocketbase/pocketbase#4195] [Q&A] [answered]
Hi all.

I am trying to save an array of objects (ideally) inside a column.  I am having issues saving any type of json object that has an array.
Is there any way to get support to have something like this inside a json column

```json
[
    {
        "event": "view_item_list",
        "ecommerce": {
            "currency": "string",
            "items": [{
                "google_business_vertical": "string",
                "id": "string",
                "index": "string",
                "item_brand": "string",
                "item_category": "string",
                "item_id": "string",
                "item_list_name": "string",
                "item_name": "string",
                "price": "string",
            }]
        }
    },
    {
        "event": "gtm4wp.productImpressionEEC",
        "ecommerce": {
            "currencyCode": "string",
            "impressions": [{
                "brand": "string",
                "category": "string",
                "id": "string",
                "list": "string",
                "name": "string",
                "position": "string",
                "price": "string",
                "stocklevel": "string|empty"
            }]
        }
    }
]
```

These objects do not contain data that the database is concerned with

**Comment by ganigeorgiev** [comment]
The above array is not a valid JSON - there is trailing comma that has to be removed after ` "price": "string",`.

**Reply by ganigeorgiev** [reply, marked as answer]
I also forgot to mention that if you are using the JS SDK and the above json is submitted together with File/Blob fields as plain body object (aka. not explicit `FormData`), make sure that you are using the latest JS SDK version (_note that there are 2 edge cases mentioned in the [JS SDK v0.20.3 release notes](https://github.com/pocketbase/js-sdk/releases/tag/v0.20.3) that are already resolved in the `develop` branch and will be available with the PocketBase v0.21.0 release soon_).

If neither removing the above trailing comma nor updating the JS SDK solves your problem, please elaborate what do you mean by _"I am having issues saving any type of json object"_ (is there an API error or the array is not saved properly?) and ideally provide an example how you are sending the request.
